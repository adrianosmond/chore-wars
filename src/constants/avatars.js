export const avatarOptions = {
  skinColor: [
    'Pale',
    'Light',
    'Tanned',
    'Yellow',
    'Brown',
    'DarkBrown',
    'Black',
  ],
  topType: [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairBigHair',
    'LongHairBob',
    'LongHairBun',
    'LongHairCurly',
    'LongHairCurvy',
    'LongHairDreads',
    'LongHairFrida',
    'LongHairFro',
    'LongHairFroBand',
    'LongHairNotTooLong',
    'LongHairShavedSides',
    'LongHairMiaWallace',
    'LongHairStraight',
    'LongHairStraight2',
    'LongHairStraightStrand',
    'ShortHairDreads01',
    'ShortHairDreads02',
    'ShortHairFrizzle',
    'ShortHairShaggyMullet',
    'ShortHairShortCurly',
    'ShortHairShortFlat',
    'ShortHairShortRound',
    'ShortHairShortWaved',
    'ShortHairSides',
    'ShortHairTheCaesar',
    'ShortHairTheCaesarSidePart',
  ],
  hatColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
  hairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'PastelPink',
    'Platinum',
    'Red',
    'SilverGray',
  ],
  facialHairType: [
    'Blank',
    'BeardMedium',
    'BeardLight',
    'BeardMagestic',
    'MoustacheFancy',
    'MoustacheMagnum',
  ],
  facialHairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'Platinum',
    'Red',
  ],
  accessoriesType: [
    'Blank',
    'Kurt',
    'Prescription01',
    'Prescription02',
    'Round',
    'Sunglasses',
    'Wayfarers',
  ],
  clotheType: [
    'BlazerShirt',
    'BlazerSweater',
    'CollarSweater',
    'Hoodie',
    'Overall',
    'ShirtCrewNeck',
    'ShirtScoopNeck',
    'ShirtVNeck',
  ],
  clotheColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
};


export const editorOrder = [
  'skinColor',
  'topType',
  'hairColor',
  'hatColor',
  'facialHairType',
  'facialHairColor',
  'accessoriesType',
  'clotheType',
  'clotheColor',
];

export const labels = {
  skinColor: 'Skin Colour',
  topType: 'Hair or Headware',
  hairColor: 'Hair Colour',
  hatColor: 'Headware Colour',
  facialHairType: 'Facial Hair Type',
  facialHairColor: 'Facial Hair Colour',
  accessoriesType: 'Glasses',
  clotheType: 'Clothing Type',
  clotheColor: 'Clothing Colour',
};

export const isEditable = {
  skinColor: () => true,
  topType: () => true,
  hairColor: options => [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairFrida',
    'LongHairShavedSides',
  ].indexOf(options.topType) < 0,
  hatColor: options => [
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
  ].indexOf(options.topType) >= 0,
  facialHairType: options => options.topType !== 'Hijab',
  facialHairColor: options => isEditable.facialHairType(options) && options.facialHairType !== 'Blank',
  accessoriesType: () => true,
  clotheType: () => true,
  clotheColor: options => !options.clotheType.startsWith('Blazer'),
};

export const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};
