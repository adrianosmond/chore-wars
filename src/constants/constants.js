const TIME_UNIT = 1000 * 60 * 60 * 24; // 1 day
const MIN_CHORE_FREQUENCY = 1;
const MAX_CHORE_FREQUENCY = 365;
const MIN_CHORE_POINTS = 1;
const MAX_CHORE_POINTS = 200;

const DATE_FORMAT = 'DD/MM/YYYY';

const MAX_POINT_DIFFERENCE = 500;

const MAX_NAME_LENGTH = 10;
const JOIN_CODE_LENGTH = 8;

const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};

const ActionTypes = {
  addChore: 'ADD_CHORE',
  resetChoreDoneDate: 'RESET_CHORE_DONE_DATE',
  blockChore: 'BLOCK_CHORE',
  unblockChore: 'UNBLOCK_CHORE',
  removeChore: 'REMOVE_CHORE',
  updateChore: 'UPDATE_CHORE',
  setChores: 'SET_CHORES',
  makeChain: 'MAKE_CHAIN',
  breakChain: 'BREAK_CHAIN',
  addPoints: 'ADD_POINTS',
  claimPrize: 'CLAIM_PRIZE',
  paidDebt: 'PAID_DEBT',
  setPoints: 'SET_POINTS',
  updateUser: 'UPDATE_USER',
  setAuthUser: 'SET_AUTH_USER',
  setGame: 'SET_GAME',
  setPointsLoaded: 'SET_POINTS_LOADED',
  setChoresLoaded: 'SET_CHORES_LOADED',
  saveStatePostUndo: 'SAVE_STATE_POST_UNDO',
};

export {
  TIME_UNIT,
  MIN_CHORE_FREQUENCY,
  MAX_CHORE_FREQUENCY,
  MIN_CHORE_POINTS,
  MAX_CHORE_POINTS,
  MAX_NAME_LENGTH,
  JOIN_CODE_LENGTH,
  DATE_FORMAT,
  MAX_POINT_DIFFERENCE,
  ActionTypes,
  DefaultAvatar,
};
