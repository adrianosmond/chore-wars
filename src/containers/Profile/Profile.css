.profile-editor__avatar {
  margin-top: -1rem;
  margin-left: -1rem;
  margin-right: -1rem;
  display: flex;
  justify-content: center;
  /* position: sticky; */
  z-index: 1;
  top: 0;
  padding: 1rem;
  background-color: #fff;
}

.profile-editor__field {
  margin-top: 1rem;
}

@media screen and (min-width: 37.5em) {
  .profile-editor {
    display: grid;
    grid-template-columns: 1fr  1fr;
    grid-column-gap: 1rem;
    align-items: flex-start;
  }
  
  .profile-editor__avatar {
    margin-top: 0;
    margin-left: 0;
    margin-right: 0;
  }
}
