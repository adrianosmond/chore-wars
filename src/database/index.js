import app from 'firebase/app';
import 'firebase/database';
import 'firebase/auth';

const config = {
  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.REACT_APP_FIREBASE_DATABASE_URL,
  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGE_SENDER_ID,
  appId: process.env.REACT_APP_FIREBASE_APP_ID,
};

app.initializeApp(config);

export default app;

export const database = app.database();
export const auth = app.auth();
export const emailCredential = app.auth.EmailAuthProvider.credential;