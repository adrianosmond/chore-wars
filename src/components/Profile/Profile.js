import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Avatar from 'avataaars';

import { setPlayerName, savePlayerName, savePlayerAvatar } from 'actions/playerActions';
import { startHoliday, stopHoliday, signOut, copyDummyData } from 'actions/sessionActions';

import { editorOrder, isEditable, labels } from 'constants/avatars';
import * as routes from 'constants/routes';
import { MAX_NAME_LENGTH } from 'constants/constants';

import './Profile.css';

const EditAvatarLinks = ({ avatar }) =>
  <div>
  { Object.keys(avatar)
      .sort((a, b) => editorOrder.indexOf(a) - editorOrder.indexOf(b))
      .map((key) => {
        if (editorOrder.indexOf(key) < 0 || !isEditable[key](avatar)) {
          return null;
        }
        return <Link to={`${routes.EDIT_PROFILE}/${key}`}className="form__button form__button--tertiary" key={key}>{labels[key]}</Link>;
      }) }
  </div>;

class Profile extends Component {
  constructor(props) {
    super(props);
    const { avatar, name } = props.player;
    this.state = {
      name,
      avatar,
    };
  }

  savePlayer(event) {
    const { user, gameId } = this.props;
    const { avatar, name } = this.state;
    const trimmedName = name.trim();
    if (trimmedName.length > 0 && trimmedName.length <= MAX_NAME_LENGTH) {
      this.props.setPlayerName(user, name.trim());
      this.props.savePlayerName(user, name.trim(), gameId);
      this.props.savePlayerAvatar(user, avatar, gameId);
    } else {
      event.preventDefault();
      // TODO: validation
    }
  }

  render() {
    const { avatar } = this.state;
    const { gameId, holidayStartTime } = this.props;
    if (!avatar) return null;
    return (
      <div className="profile-editor">
        <div className="profile-editor__column">
          <div className="profile-editor__avatar">
            <Avatar { ...avatar } />
          </div>
          <div className="profile-editor__field">
            <label htmlFor="profile-name">Name:</label>
            <input className="form__input" id="profile-name" placeholder="Name" type="text"
              maxLength={MAX_NAME_LENGTH}
              onChange={event => this.setState({ name: event.target.value })}
              value={this.state.name} />
            <div className="form__button-holder form__button-holder--tight">
              <Link to={routes.CHORES} className="form__button form__button--secondary">Cancel</Link>
              <Link to={routes.CHORES} className="form__button form__button"
                id="profile-save-player" onClick={this.savePlayer.bind(this)}>Save</Link>
            </div>
            {
              process.env.NODE_ENV === 'development' ?
              <button onClick={() => { this.props.copyData(); }}
                className="form__button form__button--secondary">Clone Data</button>
              : null
            }
            {
              holidayStartTime ?
                <button onClick={() => {
                  this.props.stopHoliday(gameId, holidayStartTime, new Date().getTime());
                }} className="form__button form__button--tertiary">Stop holiday</button>
              :
                <button onClick={() => {
                  this.props.startHoliday(gameId, new Date().getTime());
                }} className="form__button form__button--tertiary">Start holiday</button>
            }
            <button onClick={() => { this.props.doSignOut(); }}
              id="profile-sign-out"
              className="form__button form__button--tertiary">Sign out</button>
          </div>
        </div>
        <div className="profile-editor__column">
          <p className="profile-editor__field">Appearance:</p>
          <EditAvatarLinks avatar={avatar} />
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  gameId: state.session.game.gameId,
  holidayStartTime: state.session.holiday,
  user: state.session.authUser.uid,
  player: state.players[state.session.authUser.uid],
});

const mapDispatchToProps = dispatch => ({
  setPlayerName: (player, name) => dispatch(setPlayerName(player, name)),
  savePlayerName: (player, name, game) => dispatch(savePlayerName(player, name, game)),
  savePlayerAvatar: (player, avatar, game) => dispatch(savePlayerAvatar(player, avatar, game)),
  startHoliday: (game, startTime) => dispatch(startHoliday(game, startTime)),
  stopHoliday: (game, startTime, endTime) => dispatch(stopHoliday(game, startTime, endTime)),
  doSignOut: () => dispatch(signOut()),
  copyData: () => dispatch(copyDummyData()),
});

export { EditAvatarLinks };

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
