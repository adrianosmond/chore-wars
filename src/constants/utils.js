import { TIME_UNIT } from './constants'
const sortByCurrentPoints = (a, b) => b.currentPoints - a.currentPoints

const processChore = (chore) => {
  const timeSinceChore = (new Date().getTime() - chore.lastDone) / TIME_UNIT
  const timeRemaining = timeSinceChore - chore.frequency
  const percentage = Math.min(100 * timeSinceChore / chore.frequency, 100)
  const multiplier = 1 + (1 / chore.frequency)
  const due = chore.lastDone + (chore.frequency * TIME_UNIT)
  return {
    currentPoints: Math.round(percentage / 100 * chore.pointsPerTime * Math.pow(multiplier, timeRemaining)),
    percentage,
    due
  }
}

export { sortByCurrentPoints, processChore }
