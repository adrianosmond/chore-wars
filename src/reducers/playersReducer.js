import { ActionTypes } from 'constants/constants';

export default function playersReducer(state = { }, action) {
  const newState = JSON.parse(JSON.stringify(state));
  const { player } = action;

  switch (action.type) {
    case ActionTypes.setPlayers:
      return {
        ...action.players,
      };

    case ActionTypes.setPlayerName:
      newState[player].name = action.name;
      return newState;

    case ActionTypes.setPlayerAvatar:
      newState[player].avatar = action.avatar;
      return newState;

    default:
      return state;
  }
}
