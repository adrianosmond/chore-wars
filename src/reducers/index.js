import { combineReducers } from 'redux';
import undoable, { includeAction } from 'redux-undo';
import { ActionTypes } from 'constants/constants';
import choresReducer from './choresReducer';
import playersReducer from './playersReducer';
import pointsReducer from './pointsReducer';
import sessionReducer from './sessionReducer';

const undoableActions = [
  ActionTypes.addChore,
  ActionTypes.resetChoreDoneDate,
  ActionTypes.removeChore,
  ActionTypes.updateChore,
  ActionTypes.makeChain,
  ActionTypes.breakChain,
  ActionTypes.claimPrize,
  ActionTypes.paidDebt,
];

export default combineReducers({
  chores: undoable(choresReducer, {
    filter: includeAction(undoableActions),
  }),
  points: undoable(pointsReducer, {
    filter: includeAction(undoableActions),
  }),
  players: playersReducer,
  session: sessionReducer,
});
