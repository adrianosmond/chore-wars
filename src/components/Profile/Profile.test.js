import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { DefaultAvatar } from 'constants/avatars';
import { auth } from 'utils/database';
import * as playerActions from 'state/reducers/playersReducer';
import Profile, { EditAvatarLinks } from './Profile';

const mockStore = configureMockStore([thunk]);

describe('Profile', () => {
  const playerId = 'player1';
  const gameId = 'fake-game';

  const store = mockStore({
    session: {
      game: {
        gameId,
      },
      authUser: {
        uid: playerId,
      },
    },
    players: {
      [playerId]: {
        name: 'Player 1',
        avatar: DefaultAvatar,
      },
    },
  });
  const component = shallow(<Profile store={store} />).dive();
  it('Renders', () => {
    expect(component).toMatchSnapshot();
  });

  it('Can sign out', () => {
    jest.spyOn(auth, 'signOut');
    const signOutButton = component.find('#profile-sign-out');
    expect(auth.signOut).not.toHaveBeenCalled();
    signOutButton.simulate('click');
    expect(auth.signOut).toHaveBeenCalledTimes(1);
  });

  it('Can update the player name', () => {
    const newName = 'Jeff';
    const nameInput = component.find('#profile-name');
    nameInput.simulate('change', { target: { value: newName } });
    expect(component.state('name')).toBe(newName);
  });

  describe('Saving', () => {
    const saveButton = component.find('#profile-save-player');
    jest.spyOn(playerActions, 'setPlayerName');
    jest.spyOn(playerActions, 'savePlayerName');
    jest.spyOn(playerActions, 'savePlayerAvatar');

    beforeEach(() => {
      playerActions.setPlayerName.mockClear();
      playerActions.savePlayerName.mockClear();
      playerActions.savePlayerAvatar.mockClear();
    });

    it('Can save', () => {
      const newName = 'New Name';
      component.setState({
        name: newName,
      });
      expect(playerActions.setPlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerAvatar).not.toHaveBeenCalled();
      saveButton.simulate('click');
      expect(playerActions.setPlayerName).toHaveBeenCalledWith(playerId, newName);
      expect(playerActions.savePlayerName).toHaveBeenCalledWith(playerId, newName, gameId);
      expect(playerActions.savePlayerAvatar).toHaveBeenCalledTimes(1);
    });

    it('Will not save an empty name', () => {
      component.setState({
        name: '',
      });
      const preventDefault = jest.fn();
      expect(playerActions.setPlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerAvatar).not.toHaveBeenCalled();
      saveButton.simulate('click', { preventDefault });
      expect(playerActions.setPlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerAvatar).not.toHaveBeenCalled();
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it('Will not save a name that is more than MAX_LENGTH', () => {
      component.setState({
        name: 'This is a name that is far too long',
      });
      const preventDefault = jest.fn();
      expect(playerActions.setPlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerAvatar).not.toHaveBeenCalled();
      saveButton.simulate('click', { preventDefault });
      expect(playerActions.setPlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerName).not.toHaveBeenCalled();
      expect(playerActions.savePlayerAvatar).not.toHaveBeenCalled();
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });
  });

  describe('EditAvatarLinks', () => {
    it('Renders', () => {
      expect(shallow(<EditAvatarLinks avatar={DefaultAvatar} />)).toMatchSnapshot();
    });
  });
});
