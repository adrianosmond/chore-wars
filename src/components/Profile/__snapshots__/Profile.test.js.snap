// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile EditAvatarLinks Renders 1`] = `
<div>
  <Link
    className="form__button form__button--tertiary"
    key="skinColor"
    replace={false}
    to="/edit-profile/skinColor"
  >
    Skin Colour
  </Link>
  <Link
    className="form__button form__button--tertiary"
    key="topType"
    replace={false}
    to="/edit-profile/topType"
  >
    Hair or Headware
  </Link>
  <Link
    className="form__button form__button--tertiary"
    key="hairColor"
    replace={false}
    to="/edit-profile/hairColor"
  >
    Hair Colour
  </Link>
  <Link
    className="form__button form__button--tertiary"
    key="facialHairType"
    replace={false}
    to="/edit-profile/facialHairType"
  >
    Facial Hair Type
  </Link>
  <Link
    className="form__button form__button--tertiary"
    key="accessoriesType"
    replace={false}
    to="/edit-profile/accessoriesType"
  >
    Glasses
  </Link>
  <Link
    className="form__button form__button--tertiary"
    key="clotheType"
    replace={false}
    to="/edit-profile/clotheType"
  >
    Clothing Type
  </Link>
</div>
`;

exports[`Profile Renders 1`] = `
<div
  className="profile-editor"
>
  <div
    className="profile-editor__column"
  >
    <div
      className="profile-editor__avatar"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        topType="LongHairStraight"
      />
    </div>
    <div
      className="profile-editor__field"
    >
      <label
        htmlFor="profile-name"
      >
        Name:
      </label>
      <input
        className="form__input"
        id="profile-name"
        maxLength={10}
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value="Player 1"
      />
      <div
        className="form__button-holder form__button-holder--tight"
      >
        <Link
          className="form__button form__button--secondary"
          replace={false}
          to="/chores"
        >
          Cancel
        </Link>
        <Link
          className="form__button form__button"
          id="profile-save-player"
          onClick={[Function]}
          replace={false}
          to="/chores"
        >
          Save
        </Link>
      </div>
      <button
        className="form__button form__button--tertiary"
        id="profile-sign-out"
        onClick={[Function]}
      >
        Sign out
      </button>
    </div>
  </div>
  <div
    className="profile-editor__column"
  >
    <p
      className="profile-editor__field"
    >
      Appearance:
    </p>
    <EditAvatarLinks
      avatar={
        Object {
          "accessoriesType": "Blank",
          "avatarStyle": "Transparent",
          "clotheColor": "Blue03",
          "clotheType": "BlazerShirt",
          "eyeType": "Default",
          "eyebrowType": "Default",
          "facialHairColor": "BrownDark",
          "facialHairType": "Blank",
          "hairColor": "BrownDark",
          "mouthType": "Twinkle",
          "skinColor": "Light",
          "topType": "LongHairStraight",
        }
      }
    />
  </div>
</div>
`;
